packages/grid/x-data-grid-pro/src/hooks/features/columnResize/useGridColumnResize.tsx

```diff
@@ -119,18 +134,146 @@ function getResizeDirection(element: HTMLElement, direction: Direction) {
   return side;
 }
 
+function preventClick(event: MouseEvent) {
+  event.preventDefault();
+  event.stopImmediatePropagation();
+}
+
+/**
+ * Checker that returns a promise that resolves when the column virtualization
+ * is disabled.
+ */
+function useColumnVirtualizationDisabled(apiRef: React.MutableRefObject<GridPrivateApiPro>) {
+  const promise = React.useRef<ControllablePromise>();
+  const selector = () => gridVirtualizationColumnEnabledSelector(apiRef);
+  const value = useGridSelector(apiRef, selector);
+
+  React.useEffect(() => {
+    if (promise.current && value === false) {
+      promise.current.resolve();
+      promise.current = undefined;
+    }
+  });
+
+  const asyncCheck = () => {
+    if (!promise.current) {
+      if (selector() === false) {
+        return Promise.resolve();
+      }
+      promise.current = createControllablePromise();
+    }
+    return promise.current;
+  };
+
+  return asyncCheck;
+}
+
+/**
+ * Basic statistical outlier detection, checks if the value is `F * IQR` away from
+ * the Q1 and Q3 boundaries. IQR: interquartile range.
+ */
+function excludeOutliers(inputValues: number[], factor: number) {
+  if (inputValues.length < 4) {
+    return inputValues;
+  }
+
+  const values = inputValues.slice();
+  values.sort((a, b) => a - b);
+
+  const q1 = values[Math.floor(values.length * 0.25)];
+  const q3 = values[Math.floor(values.length * 0.75) - 1];
+  const iqr = q3 - q1;
+
+  // We make a small adjustment if `iqr < 5` for the cases where the IQR is
+  // very small (e.g. zero) due to very close by values in the input data.
+  // Otherwise, with an IQR of `0`, anything outside that would be considered
+  // an outlier, but it makes more sense visually to allow for this 5px variance
+  // rather than showing a cropped cell.
+  const deviation = iqr < 5 ? 5 : iqr * factor;
+
+  return values.filter((v) => v > q1 - deviation && v < q3 + deviation);
+}
+
+function extractColumnWidths(
+  apiRef: React.MutableRefObject<GridPrivateApiPro>,
+  options: AutosizeOptionsRequired,
+) {
+  type Result = Record<string, number>;
+
+  const headers = apiRef.current.columnHeadersContainerElementRef?.current!;
+  const container = apiRef.current.virtualScrollerRef?.current!;
+
+  const getHeader = (field: string) =>
+    headers.querySelector(`:scope > div > div > [data-field="${field}"][role="columnheader"]`);
+
+  const getCells = (field: string) =>
+    Array.from(
+      container.querySelectorAll(`:scope > div > div > div > [data-field="${field}"][role="cell"]`),
+    );
+
+  const widthByField = options.columns.reduce((result, column) => {
+    const cells = getCells(column.field);
+
+    const widths = cells.map((cell) => {
+      const style = window.getComputedStyle(cell, null);
+      const paddingWidth = parseInt(style.paddingLeft, 10) + parseInt(style.paddingRight, 10);
+      const contentWidth = (cell.firstElementChild?.scrollWidth ?? -1) + 1;
```

Comment:
When used with dynamic row height, the column width is increased by 1px: https://codesandbox.io/s/zen-monad-xyxwm4
It would be great to avoid that 🙂
It's also worth mentioning that column autosizing does nothing when dynamic row height is being used.